.github/workflows/e2e-hotel.yml

```diff
@@ -0,0 +1,151 @@
+      #   run: |
+      #     PROTOC_ZIP=protoc-3.14.0-linux-x86_64
+      #     curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/$PROTOC_ZIP.zip
+      #     unzip -o $PROTOC_ZIP.zip -d $HOME/.local
+      #     chmod +x $HOME/.local/bin/protoc
+      #     rm -f $PROTOC_ZIP.zip
+
+      - name: Build and push
+        working-directory: benchmarks/hotel-app
+        env:
+          GOPRIVATE_KEY: ${{ secrets.EASE_LAB_ACCESS_KEY }}
+        run: |
+          make all
+          make push
+
+  test-compose:
+    name: Test Docker Compose
+    needs: build-and-push
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        file:
+          [
+            dc-geo.yaml,
+            dc-geo-tracing.yaml,
+          ]
+
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+        with:
+          lfs: "true"
+
+      - name: start docker-compose benchmark
+        working-directory: benchmarks/hotel-app/yamls/docker-compose
+        run: |
+          docker-compose -f ${{ matrix.file }} pull
+          docker-compose -f ${{ matrix.file }} up &> log_file &
+          sleep 60s
+          cat log_file
+
+      - name: invoke the chain
+        run: |
+          ./tools/bin/grpcurl -plaintext -import-path ./tools/invoker/proto -proto helloworld.proto localhost:50000 helloworld.Greeter.SayHello
+
+      - name: show docker-compose log
+        run: cat benchmarks/hotel-app/yamls/docker-compose/log_file
+
+  test-knative:
+    name: Test Knative Deployment
+    needs: build-and-push
+    env:
+      GOCACHE: /root/tmp/gocache
+      GOPATH: /root/tmp/gopath
+    runs-on: [self-hosted, stock-knative]
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          lfs: "true"
+      - name: Checkout LFS objects
+        run: git lfs checkout
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.16
+
+      - name: Deploy knative
+        working-directory: benchmarks
+        run: ../tools/kn_deploy.sh ./hotel-app/yamls/knative/*
+
+      - name: Test invoking once
+        working-directory: tools/test-client
+        run: |
+          set -x
+          go build ./test-client.go
+
+          NODEPORT=$(kubectl get svc kourier-ingress -n kourier-system -o=jsonpath='{.spec.ports[0].nodePort}')
+
+          url_list=$(kn service list | awk '$2 ~ /http/ {sub(/http\:\/\//,""); print $2}')
+          for url in $url_list
+          do
+            ./test-client --addr $url:$NODEPORT
+          done
+
+      - name: Print logs
+        if: ${{ always() }}
+        run: |
+          set -x
+          container_list=$(kubectl get pods -n default -o jsonpath="{.items[*].spec.containers[*].name}")
+          for container_name in $container_list
+          do
+            kubectl logs -n default -c $container_name -l serving.knative.dev/service=${{ matrix.service }}
```

Comment:
And then this log functionality will work.

Reply:
resolved?